diff --git a/ckan/lib/dictization/model_dictize.py b/ckan/lib/dictization/model_dictize.py
index 9f17cf4..875e0be 100644
--- a/ckan/lib/dictization/model_dictize.py
+++ b/ckan/lib/dictization/model_dictize.py
@@ -377,7 +377,7 @@ def group_dictize(group, context):
             context['ignore_capacity_check'] = True
 
     if include_datasets:
-        q['rows'] = 1000    # Only the first 1000 datasets are returned
+        q['rows'] = 10    # Only the first 10 datasets are returned
 
     context_ = dict((k, v) for (k, v) in context.items() if k != 'schema')
     search_results = logic.get_action('package_search')(context_, q)

