diff --git a/ckan/logic/action/create.py b/ckan/logic/action/create.py
index 1a124f4..6048e6d 100644
--- a/ckan/logic/action/create.py
+++ b/ckan/logic/action/create.py
@@ -305,7 +305,13 @@ def resource_create(context, data_dict):
         _get_action('package_update')(context, pkg_dict)
         context.pop('defer_commit')
     except ValidationError, e:
-        errors = e.error_dict['resources'][-1]
+        try:
+            errors = e.error_dict['resources'][-1]
+        except KeyError, ke:
+            for item, val in e.error_summary.items():
+                ckan.lib.helpers.flash_error("%s: %s" % (item, val))
+            raise ke
+            # errors = e.error_dict
         raise ValidationError(errors)
 
     ## Get out resource_id resource from model as it will not appear in
diff --git a/ckan/logic/action/delete.py b/ckan/logic/action/delete.py
index dd56220..7b9d4b4 100644
--- a/ckan/logic/action/delete.py
+++ b/ckan/logic/action/delete.py
@@ -106,7 +106,12 @@ def resource_delete(context, data_dict):
     try:
         pkg_dict = _get_action('package_update')(context, pkg_dict)
     except ValidationError, e:
-        errors = e.error_dict['resources'][-1]
+        try:
+            errors = e.error_dict['resources'][-1]
+        except KeyError, ke:
+            for item, val in e.error_summary.items():
+                ckan.lib.helpers.flash_error("%s: %s" % (item, val))
+            raise ke
         raise ValidationError(errors)
 
     for plugin in plugins.PluginImplementations(plugins.IResourceController):
diff --git a/ckan/logic/action/update.py b/ckan/logic/action/update.py
index f87b81a..a6b0a0b 100644
--- a/ckan/logic/action/update.py
+++ b/ckan/logic/action/update.py
@@ -158,7 +158,12 @@ def resource_update(context, data_dict):
         updated_pkg_dict = _get_action('package_update')(context, pkg_dict)
         context.pop('defer_commit')
     except ValidationError, e:
-        errors = e.error_dict['resources'][n]
+        try:
+            errors = e.error_dict['resources'][n]
+        except KeyError, ke:
+            for item, val in e.error_summary.items():
+                ckan.lib.helpers.flash_error("%s: %s" % (item, val))
+            raise ke
         raise ValidationError(errors)
 
     upload.upload(id, uploader.get_max_resource_size())
