diff --git a/ckan/logic/action/create.py b/ckan/logic/action/create.py
index 1a124f4..4cef253 100644
--- a/ckan/logic/action/create.py
+++ b/ckan/logic/action/create.py
@@ -305,7 +305,11 @@ def resource_create(context, data_dict):
         _get_action('package_update')(context, pkg_dict)
         context.pop('defer_commit')
     except ValidationError, e:
-        errors = e.error_dict['resources'][-1]
+        try:
+            errors = e.error_dict['resources'][-1]
+        except KeyError:
+            errors = e.error_dict
+            errors['parent_package'] = [u"Parent dataset is not valid"]
         raise ValidationError(errors)
 
     ## Get out resource_id resource from model as it will not appear in
